pkg/appmgmt/interfaces/amprotocol.go

```diff
@@ -0,0 +1,83 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package interfaces
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+// app management protocol defines all the APIs needed for app management,
+// this is the protocol between scheduler cache and app management plugins
+type ApplicationManagementProtocol interface {
+	// returns app that already existed in the cache,
+	// or nil, false if app with the given appID is not found
+	GetApplication(appID string) (ManagedApp, bool)
```

Comment:
it makes sense, bool is not needed.

```
Sometimes you need to distinguish a missing entry from a zero value. Is there an entry for "UTC" or is that 0 because it's not in the map at all? You can discriminate with a form of multiple assignment.

var seconds int
var ok bool
seconds, ok = timeZone[tz]
For obvious reasons this is called the “comma ok” idiom. 
```